### ISSUE TITLE
Dropbox app tokens expire

### ISSUE THREAD

**Issue by louisabraham** [fsantini/KoboCloud#131]
The OAuth tokens that are supposed not to expire do expire. See https://dropbox.tech/developers/migrating-app-permissions-and-access-tokens

I need to implement a fix that stores a refresh token instead.

**Comment by louisabraham** [comment]
I investigated the issue.

Basically, now we need 3 steps:


1. Request a PKCE code
`https://www.dropbox.com/oauth2/authorize?client_id=5oyw72cfwcp352f&response_type=code&code_challenge_method=plain&code_challenge=0000000000000000000000000000000000000000000&redirect_uri=https://louisabraham.github.io/KoboCloud&token_access_type=offline`
2. Use that code to get a refresh token
```
curl https://api.dropbox.com/oauth2/token \
    -d code=<code> \
    -d grant_type=authorization_code \
    -d code_verifier=0000000000000000000000000000000000000000000 \
    -d redirect_uri=https://louisabraham.github.io/KoboCloud \
    -d client_id=5oyw72cfwcp352f 
```
3. Get an access token with the refresh_token
```
curl https://api.dropbox.com/oauth2/token \
    -d grant_type=refresh_token \
    -d refresh_token=<refresh_token> \
    -d client_id=5oyw72cfwcp352f 
```

The goal is to store the refresh token in the kobocloudrc file and execute the 3rd step before any sync.

However, the problem is that the 2nd step cannot be done in the browser because of CORS reasons.

We could ask the users to get the refresh token from the command line, what do you think?

**Comment by louisabraham** [comment]
see #133

**Comment by peterjaric** [comment]
I just deleted my old kobocloud install from my Kobo Aura One (after backuping my library of course) and installed the new version to get access to the private DropBox option, but I got the same 400 error as in #135. 

So I decided to try the steps about, but I never got to the curl stage because the first link gives a 400 too. The (same) error is:

Invalid redirect_uri: "https://louisabraham.github.io/KoboCloud": It must exactly match one of the redirect URIs you've pre-configured for your app (including the path).
